analyzer.go

```diff
@@ -195,22 +195,15 @@ func (a *Analyzer) Analyze() (platform.AnalyzedMetadata, error) {
 		previousImageRef string
 		runImageRef      string
 	)
+	appMeta, previousImageRef, err = a.retrieveAppMetadata()
+	if err != nil {
+		return platform.AnalyzedMetadata{}, err
+	}
 
-	if a.PreviousImage != nil { // Previous image is optional in Platform API >= 0.7
-		if previousImageRef, err = a.getImageIdentifier(a.PreviousImage); err != nil {
-			return platform.AnalyzedMetadata{}, errors.Wrap(err, "identifying previous image")
-		}
-
-		// continue even if the label cannot be decoded
-		if err = image.DecodeLabel(a.PreviousImage, platform.LayerMetadataLabel, &appMeta); err != nil {
-			appMeta = platform.LayersMetadata{}
-		}
-
-		if err = a.SBOMRestorer.RestoreFromPrevious(a.PreviousImage, bomSHA(appMeta)); err != nil {
+	if sha := bomSHA(appMeta); sha != "" {
```

Comment:
This is going to break some tests - I know the `SBOMRestorer.RestoreFromPrevious` is mocked and expectations may need to be updated if there are cases where the `bomSHA` returns `""` in those tests.

Reply:
Fixed.